rust/rebase/src/witness/instructions.rs

```diff
@@ -0,0 +1,76 @@
+use crate::witness::{
+    dns::Claim as DnsClaim,
+    github::{Claim as GitHubClaim, Opts as GitHubOpts},
+    self_signed::{Claim as SelfSignedClaim, Opts as SelfSignedOpts},
+    twitter::{Claim as TwitterClaim, Opts as TwitterOpts},
+    witness::WitnessError,
+};
+use schemars::{schema::RootSchema, schema_for};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct Instructions {
+    pub statement: String,
+    pub signature: String,
+    pub witness: String,
+}
+
+#[derive(Clone, Deserialize, Serialize)]
+pub enum InstructionTypes {
+    #[serde(rename = "dns")]
```

Comment:
Good point, I'll refactor to that, then special case GitHub and SoundCloud.

As far as the CF Workers crate, I had M1 related issues when I gave it a shot a couple of months ago so I reused the TZProfiles approach since it worked. If I were starting over, I'd definitely try the pure Rust approach again -- the problems I was having with the WASM tool chain got solved so it would probably work by now.

Reply:
I think you mean "snake_case"?

Reply:
You can't override rename_all on the field level, so marking as resolved.